### DISCUSSION TITLE
About Bilayer Model for Spin- 1/2 Fermion

### DISCUSSION THREAD

**Discussion by YuntaiSong** [netket/netket#1636] [Q&A] [answered]
How to define half- filling bilayer Fermion models? I was confused by the argument in SpinOrbitalFermions since I would like to define the total number of spin- 1/2 Fermions rather than each spin type. For example, imaging a 4x4 bilayer square lattice, we could define the total number of electrons to be 32 rather than 16 spin-up and 16 spin- down.

 Also, I'm not sure about defining the interaction terms in such a bilayer model. The example just included the interaction between the Fermions with the same spin. So I would appreciate it if anyone would like to provide some ideas or examples about constructing spin- exchange interaction (or Heisenberg interaction) in the Fermion system.

**Comment by PhilipVinc** [comment]
Hi. To answer your questions...

 - You can either fix the total number of electrons in your system or you can fix the number of electrons per spin subsector. If you just want to set the total number of electrons, regardless of the spin, you can use `n_electrons=32` while if you want to set the number of electrons per spin subsector, you can use `n_electrons=(16, 16)`. In general, the more relaxed your constraint is, the harder the problem will be to solve.
 - To write your hamiltonian, you can use `nkx.operator.fermion.create(Hilbert, site, spin)` where spin should be a `+- 0.5` number to identify the terms that act on spin up or down.

**Reply by gcarleo** [reply]
@PhilipVinc I remember though that fermionic operators are still experimental because they cannot handle changes in relative populations of up and down? Maybe I don't recall correctly

**Reply by PhilipVinc** [reply]
Ah right... They do handle that case, but the operators themselves are transposed. I think it should not be a problem, however, if you write an hermitian hamiltonian....

**Reply by gcarleo** [reply, marked as answer]
ah ok good, then @wwhd222 as long as you can write your Hamiltonian in terms of c^dagger_i,up ; c^dagger_i,down ; c_i,up and c_i,down... then there is no problem
